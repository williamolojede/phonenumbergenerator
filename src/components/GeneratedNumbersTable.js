import React from 'react'

export default function GeneratedNumbersTable({ numbers }) {
  return (
    <table className="w-full shadow-md">
      <thead className="bg-grey-light">
        <tr className="text-grey-dark">
          <th className="py-3">COUNT</th>
          <th className="py-3">PHONE NUMBER</th>
        </tr>
      </thead>
      <tbody>
        {
          numbers.map((phoneNumber, ix) => (
            <tr key={ix} className="border-b border-grey-light text-center text-grey-darker font-bold">
              <td className="py-3">{ix + 1}</td>
              <td className="py-3">0{phoneNumber}</td>
            </tr>
          ))
        }
      </tbody>
    </table>
  )
}
