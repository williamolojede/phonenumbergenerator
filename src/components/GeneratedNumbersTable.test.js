import React from 'react'
import { mount } from 'enzyme';

import GeneratedNumbersTable from './GeneratedNumbersTable'

it('renders without crashing', () => {
  const numbers = [
    100494045,
    100547806,
    101628526,
    102369825,
    105988240
  ];
  const wrapper = mount(<GeneratedNumbersTable numbers={numbers}/>);

  expect(wrapper).toMatchSnapshot();
});