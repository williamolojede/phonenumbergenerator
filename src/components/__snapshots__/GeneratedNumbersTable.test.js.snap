// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<GeneratedNumbersTable
  numbers={
    Array [
      100494045,
      100547806,
      101628526,
      102369825,
      105988240,
    ]
  }
>
  <table
    className="w-full shadow-md"
  >
    <thead
      className="bg-grey-light"
    >
      <tr
        className="text-grey-dark"
      >
        <th
          className="py-3"
        >
          COUNT
        </th>
        <th
          className="py-3"
        >
          PHONE NUMBER
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="border-b border-grey-light text-center text-grey-darker font-bold"
        key="0"
      >
        <td
          className="py-3"
        >
          1
        </td>
        <td
          className="py-3"
        >
          0
          100494045
        </td>
      </tr>
      <tr
        className="border-b border-grey-light text-center text-grey-darker font-bold"
        key="1"
      >
        <td
          className="py-3"
        >
          2
        </td>
        <td
          className="py-3"
        >
          0
          100547806
        </td>
      </tr>
      <tr
        className="border-b border-grey-light text-center text-grey-darker font-bold"
        key="2"
      >
        <td
          className="py-3"
        >
          3
        </td>
        <td
          className="py-3"
        >
          0
          101628526
        </td>
      </tr>
      <tr
        className="border-b border-grey-light text-center text-grey-darker font-bold"
        key="3"
      >
        <td
          className="py-3"
        >
          4
        </td>
        <td
          className="py-3"
        >
          0
          102369825
        </td>
      </tr>
      <tr
        className="border-b border-grey-light text-center text-grey-darker font-bold"
        key="4"
      >
        <td
          className="py-3"
        >
          5
        </td>
        <td
          className="py-3"
        >
          0
          105988240
        </td>
      </tr>
    </tbody>
  </table>
</GeneratedNumbersTable>
`;
